src/state/brushSlice.ts

```diff
@@ -0,0 +1,22 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+export type BrushSettings = {
+  height: number;
+};
+
+const initialState: BrushSettings = {
+  height: 0,
+};
+export const brushSlice = createSlice({
+  name: 'brush',
+  initialState,
+  reducers: {
+    setBrushHeight(state, action: PayloadAction<number>) {
+      state.height = action.payload;
+    },
```

Comment:
Computing total offset with these causes some fun re-render issues since brush and legend are lower in the tree. Axes currently do the same thing

Reply:
I'm not sure how to deal with that yet. Ideally we would have some sort of notification when all children in the React tree are rendered. Does that API exist? Not sure.

Reply:
useLayoutEffect? Idk if that solves this case though

We'll be able to see better how to address it once the generator is a function component I think